pyscf/mp/mp2.py

```diff
@@ -212,40 +212,75 @@ def make_fno(mp, thresh=1e-6, pct_occ=None, nvir_act=None, t2=None):
             Threshold on NO occupation numbers. Default is 1e-6 (very conservative).
         pct_occ : float
             Percentage of total occupation number. Default is None. If present, overrides `thresh`.
+        pvir_act : float
+            Percentage of total number of virtuals (ceiling is taken to get an integer number).
+            Default is None. If present, overrides `thresh` and `pct_occ`.
         nvir_act : int
-            Number of virtual NOs to keep. Default is None. If present, overrides `thresh` and `pct_occ`.
+            Number of virtual NOs to keep. Default is None. If present, overrides `thresh`
+            `pvir_act`, and `pct_occ`.
 
     Returns:
         frozen : list or ndarray
             List of orbitals to freeze
         no_coeff : ndarray
-            Semicanonical NO coefficients in the AO basis
+            Semicanonical NO coefficients in the AO basis, in the order
+            [occ-frz, occ-act, vir-act, vir-frz]
     '''
+    log = logger.new_logger(mp)
+
     mf = mp._scf
     dm = mp.make_rdm1(t2=t2)
 
     nmo = mp.nmo
     nocc = mp.nocc
+    frozen_mask = mp.get_frozen_mask()
+    dm = dm[numpy.ix_(frozen_mask,frozen_mask)]
     n,v = numpy.linalg.eigh(dm[nocc:,nocc:])
     idx = numpy.argsort(n)[::-1]
     n,v = n[idx], v[:,idx]
+    pct_occ_sum = numpy.cumsum(n/numpy.sum(n))
+    numpy.set_printoptions(threshold=nmo)
```

Comment:
Should this be added as an option to `log` since this is about how the logging should be done and might apply to other parts of pyscf? Or is this a special case?